### ISSUE TITLE
Lab 3: LockingTest.java

### ISSUE THREAD

**Issue by luzhuomi** [sutd50043/project_24#2]
```java
  public void grabLock(TransactionId tid, PageId pid, Permissions perm,
      boolean expected) throws Exception {

    TestUtil.LockGrabber t = new TestUtil.LockGrabber(tid, pid, perm);
    t.start();

    // if we don't have the lock after TIMEOUT, we assume blocking.
    Thread.sleep(TIMEOUT);
    assertEquals(expected, t.acquired());

    // TODO(ghuo): yes, stop() is evil, but this is unit test cleanup
    t.stop(); // this is deprecated in java 20
    t.interrupt(); // use this
    t.interrupt(); // and this.
  }
```

**Comment by luzhuomi** [comment]
fixed in c817ce9

**Comment by luzhuomi** [comment]
close
